scrapy/http/request/form.py

```diff
@@ -54,10 +55,15 @@ def _urlencode(seq, enc):
     return urlencode(values, doseq=1)
 
 
+def _create_parser_from_response(response, parser_cls):
```

Comment:
@kmike in practice, it would mean using [`lxml.html.HTMLParser`](http://lxml.de/api/lxml.html.HTMLParser-class.html) instead of [lxml.etree.HTMLParser](http://lxml.de/api/lxml.etree.HTMLParser-class.html) (which is extended by the former).

I'll promote `_root` in parsel, and add a deprecated attribute for `_root` in Scrapy's subclass.